Blockzilla/BrowserViewController.swift

```diff
@@ -215,11 +219,50 @@ class BrowserViewController: UIViewController {
         super.viewDidAppear(animated)
     }
     
+    private func setupBiometrics() {
+        // TODO: Localize this
+        self.context.localizedReason = "Authenticate to continue session."
+        self.context.localizedCancelTitle = "New session"
+        
+        // Register for foreground notification to check biometric authentication
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.UIApplicationWillEnterForeground, object: nil, queue: nil) { notification in
+            var biometricError: NSError?
+
+            // Check if user is already in a cleared session
+            if !self.webViewContainer.isHidden {
+                if (Settings.getToggle(SettingsToggle.biometricLogin)) {
+                    self.displaySplashScreen()
+                    
+                    if self.context.canEvaluatePolicy(LAPolicy.deviceOwnerAuthenticationWithBiometrics, error: &biometricError) {
+                        self.context.evaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, localizedReason: self.context.localizedReason) {
+                            [unowned self] (success, authenticationError) in
+                            
+                            DispatchQueue.main.async {
+                                if success {
+                                    self.hideSplashScreen()
+                                    self.showToolbars()
+                                } else {
+                                    // Clear the browser session, as the user failed to authenticate.
+                                    self.resetBrowser()
+                                    self.hideSplashScreen()
+                                }
+                            }
+                        }
+                    } else {
+                        // Ran into an error with biometrics, so disable them and clear the browser:
+                        Settings.set(false, forToggle: SettingsToggle.biometricLogin)
+                        self.resetBrowser()
+                        self.hideSplashScreen()
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Please use early exit style and merge the boolean logic to avoid nested ifs here:
```
if webview.hidden || yourToggleSettingsCheck { return }
```